shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/receiver/ContextFactory.java

```diff
@@ -107,13 +115,26 @@ public Context createContext(MetaStoreEvent event) {
 
   private CircusTrainConfig generateConfiguration(MetaStoreEvent event) {
     String sourceMetaStoreUri = getSourceMetaStoreUri(event);
-    String replicaTableLocation = workOutReplicaLocation(event.getDatabaseName(), event.getTableName());
+
+    String replicaDatabaseName = event.getDatabaseName();
+    String replicaTableName = event.getTableName();
+
+    TableReplication tableReplication = shuntingYardTableReplications
+        .getTableReplicationsMap()
+        .get(String.join(".", event.getDatabaseName(), event.getTableName()));
+
+    if (tableReplication != null) {
+      replicaDatabaseName = tableReplication.getReplicaDatabaseName();
+      replicaTableName = tableReplication.getReplicaTableName();
```

Comment:
you need to check if either db name or table name is actually set. Use could have set only one wanting to override for instance only the db name. You need to add tests for this as well

Reply:
That can't happen because the TableReplication class has a @NotNull constraint on SourceTable and ReplicaTable which in turn have @NotBlank constraints on DbName and TableName.

Reply:
Hold on for this. Investigating more.

Reply:
OK. I see what you were saying. But I thought we decided that for now we won't support this usecase because we need to see how this feature will be used. It is possible, that a person would want to set a DB for all selected tables etc. So, for now, its going to be one db / table combination mapped to one db / table combination and the user needs to specify both.

Reply:
We might need to have a huddle to agree on this...

Reply:
So, i have made the `db-name` and `table-name` optional for `replica-table` but an incomplete configuration eg. provide `source-table` but not provide anything for `replica-table` is not acceptable.

Also, `table-replications` by itself is optional as a whole.

Reply:
OK, so if they don't specify `table-replications` they get the default behaviour which is that the db and table names from the source are used for the replica. If they *do* provide `table-replications` then we're saying they have to provide both the replica db *and* table name?

Reply:
Yes for the first statement and No for the second.

eg.

Valid 
```
table-replications:
  - source-table:
      database-name: bdp
      table-name: etl_daily_unique_visitors
    replica-table:
      table-name: etl_daily_unique_visitors_1

```
Invalid
```
table-replications:
  - source-table:
      database-name: bdp
      table-name: etl_daily_unique_visitors

```